src/main/java/net/logstash/logback/appender/AbstractLogstashTcpSocketAppender.java

```diff
@@ -1173,15 +1173,30 @@ public Duration getReconnectionDelay() {
      * @see PreferPrimaryDestinationConnectionStrategy#setSecondaryConnectionTTL(Duration)
      * @param secondaryConnectionTTL the TTL of a connection when connected to a secondary destination
      * @throws IllegalStateException if the {@link #connectionStrategy} is not a {@link PreferPrimaryDestinationConnectionStrategy}
+     * 
+     * @deprecated use {@link PreferPrimaryDestinationConnectionStrategy#setSecondaryConnectionTTL(Duration)} instead.
      */
+    @Deprecated
     public void setSecondaryConnectionTTL(Duration secondaryConnectionTTL) {
+        addWarn("Setting <secondaryConnectionTTL> on the appender is deprecated, set it on the connection strategy using <preferPrimary.secondaryConnectionTTL> instead.");
```

Comment:
`<preferPrimary.secondaryConnectionTTL>` is a little confusing, since it looks like xml, but is not the way the value would be set in xml.

Perhaps `<preferPrimary><secondaryConnectionTTL>` would be better?  I'm looking for better ideas. (?)

Reply:
I came across this issue several times already when throwing exceptions for invalid properties for instance.

I initially though that using an XML-like name was a good idea certainly if we consider that most users will use an XML file to configure their logging system. If we follow this approach then `<preferPrimary><secondaryConnectionTTL>` is certainly better.

Another option is to opt for a JavaBean-style format and do something like `"preferPrimary.secondaryConnectionTTL"`. But this looks weird to me...

In this particular case, maybe we could simply say `Setting <secondaryConnectionTTL> directly on the appender is deprecated. Instead you should set the connection strategy to <preferPrimary> and set <secondaryConnectionTTL> on it.`

What ever option we choose, we should probably review the other classes and standardise on the chosen naming convention...

Reply:
> `Setting <secondaryConnectionTTL> directly on the appender is deprecated. Instead you should explicitly set the connection strategy to <preferPrimary> and set its <secondaryConnectionTTL> property to the desired value.`

^ I like that.

> What ever option we choose, we should probably review the other classes and standardise on the chosen naming convention

Agreed.